src/peft/tuners/lora/model.py

```diff
@@ -426,12 +433,11 @@ def _check_merge_allowed(self):
 
     @staticmethod
     def _prepare_adapter_config(peft_config, model_config):
-        if peft_config.target_modules is None:
-            if model_config["model_type"] not in TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING:
-                raise ValueError("Please specify `target_modules` in `peft_config`")
-            peft_config.target_modules = set(
-                TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING[model_config["model_type"]]
-            )
+        if model_config["model_type"] not in TRANSFORMERS_MODELS_TO_LORA_TARGET_MODULES_MAPPING:
```

Comment:
Is this change also needed for X-LoRA?

Reply:
No, I don't think so. The X-LoRA model doesn't handle target modules at all, as it delegates that to the LoRA layers.